src/tests/compaction_test/src/delete_range_runner.rs

```diff
@@ -0,0 +1,647 @@
+        if epoch % 200 == 0 {
+            tokio::time::sleep(Duration::from_secs(1)).await;
+        }
+    }
+    Ok(())
+}
+
+struct NormalState {
+    storage: LocalHummockStorage,
+    cache: BTreeMap<Vec<u8>, StorageValue>,
+    table_id: TableId,
+    debug: bool,
+    epoch: u64,
+}
+
+struct DeleteRangeState {
+    inner: NormalState,
+    delete_ranges: Vec<(Bytes, Bytes)>,
+}
+
+impl DeleteRangeState {
+    async fn new(hummock: &HummockStorage, table_id: u32, epoch: u64) -> Self {
+        Self {
+            inner: NormalState::new(hummock, table_id, epoch).await,
+            delete_ranges: vec![],
+        }
+    }
+
+    fn enable_debug(&mut self) {
+        self.inner.enable_debug();
+    }
+}
+
+#[async_trait::async_trait]
+trait CheckState {
+    async fn delete_range(&mut self, left: &[u8], right: &[u8]);
+    async fn get(&self, key: &[u8]) -> Option<Bytes>;
+    async fn scan(&self, left: &[u8], right: &[u8]) -> Vec<(Bytes, Bytes)>;
+    fn insert(&mut self, key: &[u8], val: &[u8]);
+    async fn commit(&mut self, epoch: u64) -> Result<(), String>;
+}
+
+impl NormalState {
+    async fn new(hummock: &HummockStorage, table_id: u32, epoch: u64) -> Self {
+        let table_id = TableId::new(table_id);
+        let storage = hummock.new_local(table_id).await;
+        Self {
+            cache: BTreeMap::default(),
+            storage,
+            epoch,
+            table_id,
+            debug: false,
+        }
+    }
+
+    fn enable_debug(&mut self) {
+        self.debug = true;
+    }
+
+    async fn commit_impl(
+        &mut self,
+        delete_ranges: Vec<(Bytes, Bytes)>,
+        epoch: u64,
+    ) -> Result<(), String> {
+        let data = std::mem::take(&mut self.cache)
+            .into_iter()
+            .map(|(key, val)| (Bytes::from(key), val))
+            .collect_vec();
+        self.storage
+            .ingest_batch(
+                data,
+                delete_ranges,
+                WriteOptions {
+                    epoch,
+                    table_id: self.table_id,
+                },
+            )
+            .await
+            .map_err(|e| format!("{:?}", e))?;
+        self.epoch = epoch;
+        Ok(())
+    }
+
+    async fn get_from_storage(&self, key: &[u8], ignore_range_tombstone: bool) -> Option<Bytes> {
+        self.storage
+            .get(
+                key,
+                self.epoch,
+                ReadOptions {
+                    prefix_hint: None,
+                    ignore_range_tombstone,
+                    check_bloom_filter: false,
+                    retention_seconds: None,
+                    table_id: self.table_id,
+                },
+            )
+            .await
+            .unwrap()
+    }
+
+    async fn get_impl(&self, key: &[u8], ignore_range_tombstone: bool) -> Option<Bytes> {
+        if let Some(val) = self.cache.get(key) {
+            if self.debug {
+                println!("delete by cache");
```

Comment:
Should we need to print the key or range? Otherwise, why do we need this print when running the ci-test?

Reply:
I would delete it.